# GRUB_LINUX_KERNEL_GLOBS: configurable kernel selection, resynced for grub 2.14.
# The util/grub-mkconfig.in export list gained GRUB_FORCE_EFI_ALL_VIDEO in 2.14,
# breaking the old hunk context (macaroni-os/mark-issues#689).
--- a/docs/grub.texi
+++ b/docs/grub.texi
@@ -1604,6 +1604,11 @@
 Each module will be loaded as early as possible, at the start of
 @file{grub.cfg}.
 
+@item GRUB_LINUX_KERNEL_GLOBS
+This option may be set to override the list of path globs used to find Linux
+kernels. The defaults vary by architecture, and generally include both
+@file{/boot} and @file{/}.
+
 @end table
 
 The following options are still accepted for compatibility with existing
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -163,18 +163,19 @@
 }
 
 machine=`uname -m`
-case "x$machine" in
+globs="$GRUB_LINUX_KERNEL_GLOBS"
+[ -z "$globs" ] && case "x$machine" in
     xi?86 | xx86_64)
-	list=
-	for i in /boot/vmlinuz-* /vmlinuz-* /boot/kernel-* ; do
-	    if grub_file_is_not_garbage "$i" ; then list="$list $i" ; fi
-	done ;;
-    *) 
-	list=
-	for i in /boot/vmlinuz-* /boot/vmlinux-* /vmlinuz-* /vmlinux-* /boot/kernel-* ; do
-                  if grub_file_is_not_garbage "$i" ; then list="$list $i" ; fi
-	done ;;
+	globs="/boot/vmlinuz-* /vmlinuz-* /boot/kernel-*"
+	;;
+    *)
+	globs="/boot/vmlinuz-* /boot/vmlinux-* /vmlinuz-* /vmlinux-* /boot/kernel-*"
+	;;
 esac
+list=
+for i in ${globs} ; do
+    if grub_file_is_not_garbage "$i" ; then list="$list $i" ; fi
+done
 
 case "$machine" in
     i?86) GENKERNEL_ARCH="x86" ;;
--- a/util/grub-mkconfig.in
+++ b/util/grub-mkconfig.in
@@ -257,6 +257,7 @@
   GRUB_OS_PROBER_SKIP_LIST \
   GRUB_DISABLE_SUBMENU \
   GRUB_FORCE_EFI_ALL_VIDEO
+export GRUB_LINUX_KERNEL_GLOBS
 
 if test "x${grub_cfg}" != "x"; then
   rm -f "${grub_cfg}.new"
